### PR TITLE
Fixed CONDA_ENV

### PR DESCRIPTION
**Pull request by gcomoretto**

### CODE REVIEW

File: bin/rebuild [hunk 1]
```diff
@@ -73,7 +73,7 @@ save_build_env() {
   echo "Saving environment information in ${filename}"
 
   # saving information in environment name
-  echo "#environment_name: ${LSST_CONDA_ENV_NAME}" > "${filename}"
+  echo "# environment_name: ${LSST_CONDA_ENV_NAME}" > "${filename}"
```
Comment (ktlim) [inline bin/rebuild:76]: I think this is going to affect the `-f 2` in `envconfig` line 128.

File: bin/publish [hunk 2]
```diff
@@ -129,15 +129,18 @@ if [[ -n $DISTRIBTAG && "$DISTRIBTAG" != "$BUILD" ]]; then
   echo "Adding tag '$DISTRIBTAG' at the distribution server."
 
   match='EUPS distribution ([^ ]+) version list. Version 1.0'
+  # Saving as CONDA_ENV only the environemnt reference, that may be different from the one specified in settings.cfg
   # shellcheck disable=SC1117
-  sub="EUPS distribution ${DISTRIBTAG} version list. Version 1.0\n#BUILD=\1\n#CONDA_ENV=${CONDA_ENV_REF}"
+  sub="EUPS distribution ${DISTRIBTAG} version list. Version 1.0\n#BUILD=\1\n#CONDA_ENV=${LSST_CONDA_ENV_NAME#"$SPLENV_BASE_NAME"}"
```
Comment (ktlim) [inline bin/publish:134]:
```suggestion
  sub="EUPS distribution ${DISTRIBTAG} version list. Version 1.0\n#BUILD=\1\n#CONDA_ENV=${LSST_CONDA_ENV_NAME#$SPLENV_BASE_NAME-}"
```

File: bin/publish [hunk 3]
```diff
@@ -129,15 +129,18 @@ if [[ -n $DISTRIBTAG && "$DISTRIBTAG" != "$BUILD" ]]; then
...
   src_tag="${EUPS_PKGROOT}/tags/${BUILD}.list"
   dst_tag="${EUPS_PKGROOT}/tags/${DISTRIBTAG}.list"
 
   run sed -r "s|${match}|${sub}|" "$src_tag" > "$dst_tag"
 
   # remove the original bNNNN tag to reduce tag proliferation
   rm "$src_tag"
+
+  run cp "${LSSTSW_BUILD_DIR}/builds/${BUILD}.env" "${EUPS_PKGROOT}/env/${DISTRIBTAG}.env"
```
Comment (ktlim) [inline bin/publish:143]: This looks like it belongs more with line 125 (the other `cp`).  Also line 126 (`ls`) looks like more of a debugging output and can be removed.

**Review by ktlim — APPROVED** [review]

File: bin/rebuild (hunk 1 above)
Comment (gcomoretto) [reply bin/rebuild:76]: right, better to avoid this change

**Pull request merged**